x/authenticator/ante/authenticator_test.go

```diff
@@ -0,0 +1,323 @@
+		FromAddress: sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[0]),
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	testMsg2 := &banktypes.MsgSend{
+		FromAddress: sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	feeCoins := sdk.Coins{sdk.NewInt64Coin(osmoToken, 2500)}
+
+	err := s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	s.Require().NoError(err)
+
+	tx, _ := GenTx(
+		s.EncodingConfig.TxConfig,
+		[]sdk.Msg{
+			testMsg1,
+			testMsg2,
+		},
+		feeCoins,
+		300000,
+		"",
+		[]uint64{0, 0},
+		[]uint64{0, 0},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[0],
+			s.TestPrivKeys[1],
+		},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[0],
+			s.TestPrivKeys[1],
+		},
+	)
+
+	anteHandler := sdk.ChainAnteDecorators(s.AuthenticatorDecorator)
+	_, err = anteHandler(s.Ctx, tx, false)
+
+	s.Require().NoError(err)
+}
+
+// TestSignatureVerificationWithAuthenticatorInStore test out of gas error
+func (s *AutherticatorAnteSuite) TestSignatureVerificationOutOfGas() {
+	osmoToken := "osmo"
+	coins := sdk.Coins{sdk.NewInt64Coin(osmoToken, 2500)}
+
+	// Create a test messages for signing
+	testMsg1 := &banktypes.MsgSend{
+		FromAddress: sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[0]),
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	testMsg2 := &banktypes.MsgSend{
+		FromAddress: sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		ToAddress:   sdk.MustBech32ifyAddressBytes(osmoToken, s.TestAccAddress[1]),
+		Amount:      coins,
+	}
+	feeCoins := sdk.Coins{sdk.NewInt64Coin(osmoToken, 2500)}
+
+	err := s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	err = s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	err = s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[0].PubKey().Bytes(),
+	)
+	err = s.OsmosisApp.AuthenticatorKeeper.AddAuthenticator(
+		s.Ctx,
+		s.TestAccAddress[0],
+		"SignatureVerificationAuthenticator",
+		s.TestPrivKeys[1].PubKey().Bytes(),
+	)
+	s.Require().NoError(err)
+
+	tx, _ := GenTx(
+		s.EncodingConfig.TxConfig,
+		[]sdk.Msg{
+			testMsg1,
+			testMsg2,
+		},
+		feeCoins,
+		300000,
+		"",
+		[]uint64{0, 0},
+		[]uint64{0, 0},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[0],
+			s.TestPrivKeys[1],
+		},
+		[]cryptotypes.PrivKey{
+			s.TestPrivKeys[1],
+			s.TestPrivKeys[1],
+		},
+	)
```

Comment:
I find these a bit hard to read. Maybe we could do something like

```
type TxGenConfig struct {
    Gen     client.TxConfig
    FeeAmt  sdk.Coins
    Gas     uint64
    ChainID string
}

type AccountInfo struct {
    AccNums []uint64
    AccSeqs []uint64
}

type KeyInfo struct {
    Signers    []cryptotypes.PrivKey
    Signatures []cryptotypes.PrivKey
}

func GenTx(
    config TxGenConfig,
    msgs []sdk.Msg,
    accountInfo AccountInfo,
    keyInfo KeyInfo,
) (sdk.Tx, error) 
...

txConfig := TxGenConfig{
        Gen:     s.EncodingConfig.TxConfig,
        FeeAmt:  feeCoins,
        Gas:     300000,
        ChainID: "",
}

tx, _ := GenTx(
    txConfig,
    []sdk.Msg{
        testMsg1,
        testMsg2,
    },
    AccountInfo{
        AccNums: []uint64{0, 0},
        AccSeqs: []uint64{0, 0},
    },
    KeyInfo{
        Signers: []cryptotypes.PrivKey{
            s.TestPrivKeys[0],
            s.TestPrivKeys[1],
        },
        Signatures: []cryptotypes.PrivKey{
            s.TestPrivKeys[1],
            s.TestPrivKeys[1],
        },
    },
)
```

Reply:
Defo, I'll be working on more tests today, would you mind if I added this to the next PR?

Happy to add it here as well, lmk what you think?